### ISSUE TITLE
vagrant 2.4.0 rsync does not work on Windows Hosts

### ISSUE THREAD

**Issue by Jasper-Ben** [hashicorp/vagrant#13295]
### Debug output

![MicrosoftTeams-image (1)](https://github.com/hashicorp/vagrant/assets/12223583/7297fccd-369e-42b9-88b3-1553b0fed283)

### Expected behavior

rsync should work on Windows Host

### Actual behavior

rsync does not work on Windows Host. Maybe related to https://github.com/hashicorp/vagrant/issues/13284?

#### Vagrant version

2.4.0

#### Host operating system

Windows 10.0.19045.3693

#### Guest operating system

Debian 11

#### Steps to reproduce

1. configure folder rsync in vagrant file
2. run vagrant up

#### Vagrantfile

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "generic/debian11"
  config.vm.synced_folder "~/", "/home/vagrant/",
      owner: "vagrant",
      group: "vagrant",
      type: "rsync",
      rsync__args: [
        "-rh", "--verbose", 
        "--include=.gitconfig",
        "--include=.vimrc",
        "--include=.bashrc",
        "--exclude=*"
      ],
      rsync__verbose: true
  end
end
```

**Comment by guillermo-jano** [comment]
> Maybe related to #13284?

Does the `dos2unix` command I mentioned in my issue fix the problem for you? If so then probably they are related.

**Comment by Jasper-Ben** [comment]
> > Maybe related to #13284?
> 
> Does the `dos2unix` command I mentioned in my issue fix the problem for you? If so then probably they are related.

👋

(Un)fortunately I do not have a Windows Host setup ready for testing this myself. The issue in question was reported to me by a colleague, I just figured that a downgrade might fix it, which turned out to be correct.

As it stands, this issue is just to low-priority for me personally to justify the hassle of getting a windows test setup up and running.

But hopefully this open issue helps others that might run into the same issue and prevents extended debugging sessions 🙂

**Comment by msvihra** [comment]
I can confirm that `dos2unix` or changing line-endings for `.vagrant/machines/<machine_name>/virtualbox/private_key` to LF will fix problems as described in https://github.com/hashicorp/vagrant/issues/13284. This will work only with embedded SSH/Rsync and will not work with SSH on Windows host.

**Comment by Jasper-Ben** [comment]
I am going to test this again this week with 2.4.1. I assume this is fixed by https://github.com/hashicorp/vagrant/pull/13329 as well.

**Comment by Jasper-Ben** [comment]
Can confirm that this is fixed in 2.4.1. Closing :slightly_smiling_face:
